[CI] Humble build from source fails (Iron is broken too)

Humble source build fails since quite a long time, see [here](https://github.com/ros-controls/ros2_control/actions/runs/7334734856/job/19971781860#step:4:23528)

>   In file included from /home/runner/work/ros2_control/ros2_control/ros_ws/src/dc3eq9dhmh6/ros2_control/controller_interface/src/controller_interface_base.cpp:15:
>   /home/runner/work/ros2_control/ros2_control/ros_ws/src/dc3eq9dhmh6/ros2_control/controller_interface/include/controller_interface/controller_interface_base.hpp:119:70: error: ‘class rclcpp::NodeOptions’ has no member named ‘enable_logger_service’
>     119 |     const rclcpp::NodeOptions & node_options = rclcpp::NodeOptions().enable_logger_service(true));

There is [a step ](https://github.com/ros-controls/ros2_control/actions/runs/7334734856/job/19971781860#step:4:400)where it is switching to the latest commit on rolling

>   /usr/bin/bash -c vcs import --force --recursive src/ < package.repo
>   === src/dc3eq9dhmh6/ros2_control (git) ===
>   Cloning into '.'...
>   Note: switching to 'e92b6c2c84946c8dab44146a5b4a666390543d61'.

Why? Where is this package.repo coming from? On iron happens the same, but there was no API break with upstream packages yet.

--- a/.github/workflows/reusable-ros-tooling-source-build.yml
+++ b/.github/workflows/reusable-ros-tooling-source-build.yml
@@ -36,6 +36,7 @@ jobs:
         with:
           target-ros2-distro: ${{ inputs.ros_distro }}
           # build all packages listed in the meta package
+          ref: ${{ inputs.ref }} # otherwise the default branch is used for scheduled workflows
           package-name:
             controller_interface
             controller_manager

